tavern_server/src/db_tables.sql

```diff
@@ -0,0 +1,861 @@
+--enum types
+CREATE TYPE IF NOT EXISTS Gender AS ENUM (
+	'Male', 
+	'Female', 
+	'Other'
+);
+
+CREATE TYPE IF NOT EXISTS Size AS ENUM (
+	'Fine', 
+	'Diminutive', 
+	'Tiny', 
+	'Small', 
+	'Medium', 
+	'Large', 
+	'Huge', 
+	'Gargantuan', 
+	'Colossal'
+);
+
+CREATE TYPE IF NOT EXISTS Alignment AS ENUM (
+	'Lawful Good', 
+	'Lawful Neutral', 
+	'Lawful Evil', 
+	'Neutral Good', 
+	'True Neutral', 
+	'Neutral Evil', 
+	'Chaotic Good', 
+	'Chaotic Neutral', 
+	'Chaotic Evil'
+);
+
+CREATE TYPE IF NOT EXISTS Attribute AS ENUM (
+	'Strength', 
+	'Dexterity', 
+	'Constitution', 
+	'Intelligence', 
+	'Wisdom', 
+	'Charisma'
+);
+
+CREATE TYPE IF NOT EXISTS Skill AS ENUM (
+	'Acrobatics', 
+	'Appraise', 
+	'Bluff', 
+	'Climb', 
+	'Craft', 
+	'Diplomacy', 
+	'Disable Device', 
+	'Disguise', 
+	'Escape Artist', 
+	'Fly', 
+	'Handle Animal', 
+	'Heal', 
+	'Intimidate', 
+	'Knowledge (arcana)', 
+	'Knowledge (dungeoneering)', 
+	'Knowledge (engineering)', 
+	'Knowledge (geography)', 
+	'Knowledge (history)', 
+	'Knowledge (local)', 
+	'Knowledge (nobility)', 
+	'Knowledge (planes)', 
+	'Knowledge (religion)', 
+	'Linguistics', 
+	'Perception', 
+	'Perform', 
+	'Profession', 
+	'Ride', 
+	'Sense Motive', 
+	'Sleight of Hand', 
+	'Spellcraft', 
+	'Stealth', 
+	'Survival', 
+	'Swim', 
+	'Use Magic Device'
+);
+
+CREATE TYPE IF NOT EXISTS SaveThrow AS ENUM (
+	'Fortitude', 
+	'Reflex',
+	'Will'
+);
+
+CREATE TYPE IF NOT EXISTS CharacterStat AS ENUM (
+	'Name',
+	'Race',
+	'Size',
+	'Height',
+	'Weight',
+	'Age',
+	'Gender',
+	'Alignment',
+	'Deity',
+	'Languages',
+	'Appearance'
+);
+
+CREATE TYPE IF NOT EXISTS CombatStat AS ENUM (
+	'Melee Attack Bonus',
+	'Ranged Attack Bonus',
+	'CMB',
+	'CMD',
+	'Armor Class',
+	'Touch AC',
+	'Flat-Footed AC',
+	'Initiative Bonus',
+	'Damage Reduction',
+	'Spell Resistance',
+	'Speed',
+	'Fortitude Save',
+	'Reflex Save',
+	'Will Save'
+);
+
+CREATE TYPE IF NOT EXISTS RaceType AS ENUM (
+	'Abberation', 
+	'Animal', 
+	'Construct', 
+	'Dragon', 
+	'Fey', 
+	'Humanoid', 
+	'Magical Beast', 
+	'Monstrous Humanoid', 
+	'Ooze', 
+	'Outsider', 
+	'Plant', 
+	'Undead', 
+	'Vermin'
+);
+
+CREATE TYPE IF NOT EXISTS RaceSubtyep AS ENUM (
+
```

Comment:
You seem to have left this enum empty...

Reply:
Turns out this was a memory leak - a leak from my memory. I ended up putting both RaceType and RaceSubtype into tables instead of enums (similar to the handling of materials) and have removed both of these since.